### ISSUE TITLE
Consider creating the ckeditor5-builds repository

### ISSUE THREAD

**Issue by ma2ciek** [ckeditor/ckeditor5#1452]
An idea came to my mind recently that we actually don't need the repository-per-build structure.

This package could consist of the following directories:
 * classic-editor
   - sample
   - src
   - build
   - tests
 * inline-editor
   - sample
   - ...
 * ...

With the above architecture we could take advantages in the following fields:
 - reduce the number of packages that are installed with every `mgit bootstrap`
 - provide all samples at once so this would become an easier starting-point for new people
 - reduce the number of GH repositories which we need to maintain (setting badges, setting CI, licenses, etc.)

Each sub-build should be still easily published on the NPM, but I'm not sure if that would work well with Lerna and our testing env.

WDYT? (I'm not sure if that idea was written in the past, If yes I couldn't find the proper ticket)

**Comment by pomek** [comment]
> Each sub-build should be still easily published on the NPM, but I'm not sure if that would work well with Lerna and our testing env.

About Lerna – we are going to drop them – https://github.com/ckeditor/ckeditor5/pull/1345.

**Comment by oleq** [comment]
cc @Reinmar

**Comment by jodator** [comment]
>     * reduce the number of packages

:trollface: I do not think this is in line with our philosophy... 

I'm divided but it looks like fine idea.

As for the integration - couldn't builds go to the separate repository outside 'dev' environment? Similarly to the `ckeditor5-dev` is maintained. I'm rarely using `*build-*` repos (if at all) as I'd rather use associated `editor-*` package for testing stuff.

**Comment by Reinmar** [comment]
> * reduce the number of packages that are installed with every `mgit bootstrap`

Repositories, not packages. I assume you meant keeping the current package names and merging them into a monorepo. Right?

If I understood correctly, then:

* We'd have a multi-repo (ckeditor5) with a monorepo inside (ckeditor5-builds). This will require adjusting some of our tools (test runner, CI scripts). This will, in turn, complicate them because they will have to consider two types of `ckeditor5-*` repos. More `if()`s, more to maintain.
* The customisability of builds is based on the fact that they are maintained in separate repositories. You fork that single repository, which contains only the code and dependencies you'll need. Not only changing this would mean making breaking changes, but also may have the opposite effect (more code in the repo to go through, a combined list of dependencies, etc). Plus, this would require rewriting lots of docs.
* This would be a totally backward incompatible change. We'd have to create a migration guide for this.
* A single exception from the rule that package==repository may be confusing.
* This would reduce the number of repositories from 42 to 38. Not much of a gain.

In other words – I don't see us making this change now because the cost for us will be huge, the cost for the community would be significant too and, finally, I don't think it would change the situation much.

If anything, we should merge all repositories into one :trollface:
